src/scripts/Game.ts

```diff
@@ -87,9 +87,36 @@ class Game {
         Weather.generateWeather(now);
         RoamingPokemonList.generateIncreasedChanceRoutes(now);
 
+        this.computeOfflineEarnings();
+
         this.gameState = GameConstants.GameState.fighting;
     }
 
+    computeOfflineEarnings() {
+        const now = Date.now();
+        const timeDiffInSeconds = (now - player._previousStartLastSeen) / 1000;
+        if (timeDiffInSeconds > 1) {
+            const timeDiffOverride = timeDiffInSeconds > 86400 ? 86400 : timeDiffInSeconds;//The maximum value is 24h
+            const route: number = player.route();
```

Comment:
I think we should add some logic here to be idling the current regions first route if they weren't on a route when they quit the game.

Reply:
Is this a safety if the player's ctor changes ? It's handled by the player ctor, meaning it should never be invalid. I can make the change though ^^ Would you rather I turned the code in the player ctor into a free function or duplicate it for the time being ?

Reply:
I'm not sure what you mean by `ctor` sorry 😅.
The `player.route()` value changes to `0` if the player last saved in a town.
If this is the case, we should assume they idled the first route of the region, rather than no route.

Reply:
Sorry ^^" ctor = constructor.
In the player  `constructor(savedPlayer?) ` we switch the 0 value to a correct one (if needed), and my code should be executed after (as I'm using the _lastSeen that is updated some lines before). That's why I wanted to know if you wanted me to add the security you were talking about nonetheless ;) 
Sorry my comment lacked clarity :/

Reply:
I don't think we need to change the actual value that's being saved, just calculate the start route here if they aren't on a route.
Otherwise it will affect other aspects of the game.

Reply:
I'm not talking about changing the saved value but I'm talking about the code starting line 45 of Player.ts 
If I understand correctly, as soon as the player is constructed, its route and region are switched to a "correct" one. Meaning, when the part described in the comment will be executed, the route should always be correct. I've made the change though, I'll update the review soon ;)